regression-tests/sparktkregtests/testcases/frames/frame_timeseries_test.py

```diff
@@ -98,5 +100,74 @@ def test_frame_timeseries_dickey_fuller_constant_trend_squared(self):
         self.assertAlmostEqual(result.p_value, df_ctt_result[1], delta=0.0001)
         self.assertAlmostEqual(result.test_stat, df_ctt_result[0], delta=0.01)
 
+    def test_frame_timeseries_breusch_pagan(self):
```

Comment:
The auxillary regression is the term used to refer to the LinearRegression being run on the square of the residuals in the case of breusch pagan, and in the case of breusch godfrey running linear regression on the lagged residuals.

Basically I am asking you to either document the lines 119-121, 125-127, 151 165.

For a reference you can use 
http://people.stfx.ca/tleo/econ370term2lec1.pdf for breusch pagan, just say "this regression is determined according to page 5 of this reference" or something to that effect, and 
https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwjn08rdoczPAhVMz2MKHR4ZA9YQFggeMAA&url=http%3A%2F%2Fwww3.nccu.edu.tw%2F~maotai%2Ftrics%2FLect_11_2_AR_test.doc&usg=AFQjCNGSrXQs0wyYm19u18-FafxCdEMHMw&sig2=ZmpSQzwJWEVMBpzXs18Nhg page 2 for breusch godfrey, rather than trying to get into the nitty gritty and give the formulae in this code since I am not sure if there's enough room to really explain why you are running the regressions with the inputs you are.